executor/runcexecutor/executor_linux.go

```diff
@@ -0,0 +1,193 @@
+package runcexecutor
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/containerd/console"
+	"github.com/containerd/containerd"
+	runc "github.com/containerd/go-runc"
+	"github.com/docker/docker/pkg/signal"
+	"github.com/moby/buildkit/executor"
+	"github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sync/errgroup"
+)
+
+func (w *runcExecutor) run(ctx context.Context, id, bundle string, process executor.ProcessInfo) (int, error) {
+	return w.callWithIO(ctx, id, bundle, process, func(ctx context.Context, pidfile string, io runc.IO) (int, error) {
+		return w.runc.Run(ctx, id, bundle, &runc.CreateOpts{
```

Comment:
Instead of the `proc/ppid` hack could we make a upstream change so we can get the pid directly from this/exec function. A `Started chan int` field in `runc.CreateOpts` should suffice. https://github.com/containerd/go-runc/blob/master/runc.go#L267 would call

```
if opts.Started != nil {
  opts.Started <- cmd.Process.Pid
}
```

We can still merge the current and fix later when upstream has updated.

Reply:
Yeah, we will need to fix upstream.  There is another issue upstream which prevents us from discovering the exit status of `runc.Exec` when there is stdio associated.  These lines discard the status code:
https://github.com/containerd/go-runc/blob/master/runc.go#L241-L245
I will try to get these changes in, but I suspect it will be a slow process.

Reply:
I have created a PR: https://github.com/containerd/go-runc/pull/69 I will remove our local hacks when hopefully the RP is merged eventually.